### PR TITLE
Revoke keys and login

### PR DESCRIPTION
**Pull request by shubham7109**
### Description
PR to revoke keys and signins between sample sessions. 

### Links and data
Link: [#4723](https://devtopia.esri.com/runtime/kotlin/issues/4723)

### Changes made
- Added toolkit auth dependency on the root `app` library to be able to call `signOut()`. 
- `startSample` is now a suspend function 
- Removed workaround and cleanup code for Show portal user info and Create and save map.

### CODE REVIEW

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt [hunk 1]
```diff
@@ -140,10 +143,14 @@ data class Sample(
 /**
  * Starts the sample activity.
  */
-fun Sample.startSample(context: Context) {
+suspend fun Sample.startSample(context: Context) {
+    // Revoke previously configured ArcGISEnvironment settings like ApiKeys/OAuth tokens/Credentials
+    ArcGISEnvironment.authenticationManager.signOut()
+    ArcGISEnvironment.apiKey = null
```
Comment (TADraeseke) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt:149]: Testing around the app, these lines seemed to work!

**Review by TADraeseke — APPROVED** [review]
LGTM!

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt [hunk 2]
```diff
@@ -140,10 +143,14 @@ data class Sample(
 /**
  * Starts the sample activity.
  */
-fun Sample.startSample(context: Context) {
+suspend fun Sample.startSample(context: Context) {
```
Comment (gunt0001) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt:146]: Nitpick - "Sample" in the function name is redundant, could be:
```suggestion
suspend fun Sample.start(context: Context) {
```

File: samples/show-portal-user-info/src/main/java/com/esri/arcgismaps/sample/showportaluserinfo/MainActivity.kt [hunk 3]
```diff
@@ -22,19 +22,13 @@ import androidx.activity.compose.setContent
...
 import com.esri.arcgismaps.sample.sampleslib.theme.SampleAppTheme
 import com.esri.arcgismaps.sample.showportaluserinfo.screens.MainScreen
 
 class MainActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        // authentication with an API key or named user is
-        // required to access basemaps and other location services
-        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.ACCESS_TOKEN)
```
Comment (gunt0001) [inline samples/show-portal-user-info/src/main/java/com/esri/arcgismaps/sample/showportaluserinfo/MainActivity.kt:36]: Just to confirm - each sample that requires an ApiKey sets it in its own activity?

**Review by gunt0001 — APPROVED** [review]
@shubham7109 - looks good, one question.

@01smito01 - please could you review as well.

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt [hunk 4]
```diff
@@ -20,6 +20,9 @@ import android.app.Activity
 import android.content.Context
 import android.content.ContextWrapper
 import android.content.Intent
+import com.arcgismaps.ApiKey
```
Comment (01smito01) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt:23]: Think we can remove this? It's not used anywhere in the class.

File: samples/create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/MainActivity.kt [hunk 5]
```diff
@@ -23,26 +23,15 @@ import androidx.compose.material3.MaterialTheme
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        // This sample uses an ArcGIS Online login to be able to save a map as an ArcGIS portal item
-        // No need for license strings or an API key
-        ArcGISEnvironment.apiKey = null
-
-        // Sign out of any portals which are already authenticated
-        runBlocking {
-            ArcGISEnvironment.authenticationManager.signOut()
-        }
```
Comment (01smito01) [inline samples/create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/MainActivity.kt:45]: Thanks for doing this here - I can close the draft PR that's been sitting around for ages.

**Review by 01smito01 — APPROVED** [review]
LGTM - built the sample viewer and seems to be behaving as expected!

File: samples/show-portal-user-info/src/main/java/com/esri/arcgismaps/sample/showportaluserinfo/MainActivity.kt (hunk 3 above)
Comment (shubham7109) [reply samples/show-portal-user-info/src/main/java/com/esri/arcgismaps/sample/showportaluserinfo/MainActivity.kt:36]: Yes, all samples that need an API key should be set in its own activity. The samples that do not need an API key:
- ShowPortalUserInfo
- AuthenticateWithOAuth
- CreateAndSaveMap

File: samples/create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/MainActivity.kt (hunk 5 above)
Comment (shubham7109) [reply samples/create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/MainActivity.kt:45]: Thanks for creating it, I added it here to make sure the new sample viewer design contains your PR changes.

**Pull request merged**